src/app/static/src/tests/components/plots/barchart/filterSelect.test.ts

```diff
@@ -0,0 +1,114 @@
+import {createLocalVue, shallowMount} from "@vue/test-utils";
+
+const localVue = createLocalVue();
+
+const propsData = {
+    id: "test",
+    options: [{id: "fo1", label: "option 1"}, {id: "fo2", label: "option 2"}],
+    isXAxis: true,
+    isDisaggregateBy: false,
+    value: [{id: "fo2", label: "option 2"}],
+    label: "Test Filter"
+};
+
+const getWrapper = (props: any = {}) => {
+    return shallowMount(FilterSelect, {propsData: {...propsData, ...props}, localVue});
+};
+
+describe("FilterSelect component", () => {
+    it("renders as expected", () => {
+        const  wrapper = getWrapper();
+
+        const label = wrapper.find("label");
+        expect(label.text()).toBe("Test Filter");
+
+        const treeSelect = wrapper.find("tree-select-stub");
+        expect(treeSelect.attributes("instanceid")).toBe("test");
+        expect(treeSelect.attributes("multiple")).toBe("true");
+        expect(treeSelect.attributes("flat")).toBe("true");
+
+        const badge = wrapper.find("span");
+        expect(badge.text()).toBe("x axis");
+    });
+
+    it("initialises selected", () => {
+        const  wrapper = getWrapper();
+        const vm = (wrapper as any).vm;
+
+        expect(vm.selected).toStrictEqual([{id: "fo2", label: "option 2"}]);
+    });
+
+    it("emits input event on select when isXAxisOrDisAgg", () => {
+        const wrapper = getWrapper();
+        const vm = (wrapper as any).vm;
+
+        vm.select({id: "fo1", label: "option 1"});
+        expect(wrapper.emitted("input")[0][0]).toStrictEqual([
+            {id: "fo2", label: "option 2"}, {id: "fo1", label: "option 1"}]);
+    });
+
+    it("emits input event on select when not isXAxisOrDisAgg", () => {
+        const wrapper = getWrapper({isXAxis: false, isDisaggregateBy: false});
+        const vm = (wrapper as any).vm;
+
+        vm.select({id: "fo1", label: "option 1"});
+        expect(wrapper.emitted("input")[0][0]).toStrictEqual([{id: "fo1", label: "option 1"}]);
+    });
+
+    it("emits input event on deselect", () => {
+        const wrapper = getWrapper();
+        const vm = (wrapper as any).vm;
+
+        vm.deselect({id: "fo2", label: "option 2"});
+        expect(wrapper.emitted("input")[0][0]).toStrictEqual([]);
+    });
+
+    it("computes isXAxisOrDisagg", () => {
+        let wrapper = getWrapper();
+        let vm = (wrapper as any).vm;
+        expect(vm.isXAxisOrDisagg).toEqual(true);
+
+        wrapper = getWrapper({isXAxis: false, isDisaggregateBy: true});
+        vm = (wrapper as any).vm;
+        expect(vm.isXAxisOrDisagg).toEqual(true);
+
+        wrapper = getWrapper({isXAxis: false, isDisaggregateBy: false});
+        vm = (wrapper as any).vm;
+        expect(vm.isXAxisOrDisagg).toEqual(false);
+    });
+
+    it("computes selectedValues", () => {
+        const wrapper = getWrapper();
+        const vm = (wrapper as any).vm;
+        expect(vm.selectedValues).toStrictEqual(["fo2"]);
+    });
+
+    it("computes badge", () => {
```

Comment:
What about if it's both?

Reply:
So at the moment, the x-axis overrides disagg, so it will just say 'xaxis' and it won't show any disagg column. I think that's probably a reasonable behavior, since you wouldn't see any disagg groupings if you'd selected the same column to be both. Does that sound ok?
Will add a test for that too.

Reply:
cool, that makes sense to me :+1: